src/h2load.cc

```diff
@@ -252,6 +255,61 @@ void rate_period_timeout_w_cb(struct ev_loop *loop, ev_timer *w, int revents) {
 } // namespace
 
 namespace {
+  // Called when the warmup duration for infinite number of requests are over
+  void warmup_timeout_cb(struct ev_loop *loop, ev_timer *w, int revents) {
+    auto client = static_cast<Client *>(w->data);
+    
+    client->warmup = 2;
+    
+    ev_timer_stop(client->worker->loop, &client->warmup_watcher);
```

Comment:
I'm really sorry but I am not actually too much familiar with timer stuff here. They use `ev_timer_start` and `ev_timer_stop` to start and stop timers. I'll look around for the implementation strategies of one-shot timer with them.